src/assets/wise5/components/audioOscillator/audio-oscillator-student/audio-oscillator-student.component.ts

```diff
@@ -415,56 +355,70 @@ class AudioOscillatorController extends ComponentController {
     ctx.stroke();
   }
 
-  drawVerticalLines(ctx, width, height, gridCellSize) {
+  drawVerticalLines(ctx: any, width: number, height: number, gridCellSize: number): void {
     let x = 0;
     while (x < width) {
       this.drawVerticalLine(ctx, x, height);
       x += gridCellSize;
     }
   }
 
-  drawVerticalLine(ctx, x, height) {
+  drawVerticalLine(ctx: any, x: number, height: number): void {
     ctx.moveTo(x, 0);
     ctx.lineTo(x, height);
   }
 
-  drawHorizontalLines(ctx, width, height, gridCellSize) {
-    // draw the horizontal lines above and including the middle line
+  drawHorizontalLines(ctx: any, width: number, height: number, gridCellSize: number): void {
+    this.drawHorizontalLinesAboveMiddle(ctx, width, height, gridCellSize);
+    this.drawHorizontalLinesBelowMiddle(ctx, width, height, gridCellSize);
+  }
+
+  drawHorizontalLinesAboveMiddle(
+    ctx: any,
+    width: number,
+    height: number,
+    gridCellSize: number
+  ): void {
     let y = height / 2;
     while (y >= 0) {
       this.drawHorizontalLine(ctx, y, width);
       y -= gridCellSize;
     }
+  }
 
-    // draw the horizontal lines below the middle line
-    y = height / 2;
+  drawHorizontalLinesBelowMiddle(
+    ctx: any,
+    width: number,
+    height: number,
+    gridCellSize: number
+  ): void {
+    let y = height / 2;
     while (y < height) {
       this.drawHorizontalLine(ctx, y, width);
       y += gridCellSize;
     }
   }
 
-  drawHorizontalLine(ctx, y, width) {
+  drawHorizontalLine(ctx: any, y: number, width: number): void {
     ctx.moveTo(0, y);
     ctx.lineTo(width, y);
   }
 
-  oscillatorTypeChanged() {
+  oscillatorTypeChanged(): void {
```

Comment:
```oscillatorTypeChanged()``` and ```frequencyChanged()``` are similar. Can you refactor into a common function? Maybe name it something that resembles what the function does, like ```redraw()``` or something of that sort?